cli/smartcontract/generate.go

```diff
@@ -0,0 +1,348 @@
+Common packages from interop can be aliases with "storage", "iterator" or "ledger".`,
+		},
+		cli.StringSliceFlag{
+			Name: "callflag",
+			Usage: `Override call flags. Should have form <method>=<callflag>.
+Call flag can be one of ` + strings.Join(callFlags, ","),
+		},
+	},
+}
+
+func scTypeToGo(typ smartcontract.ParamType) string {
+	switch typ {
+	case smartcontract.AnyType:
+		return "interface{}"
+	case smartcontract.BoolType:
+		return "bool"
+	case smartcontract.IntegerType:
+		return "int"
+	case smartcontract.ByteArrayType:
+		return "[]byte"
+	case smartcontract.StringType:
+		return "string"
+	case smartcontract.Hash160Type:
+		return "interop.Hash160"
+	case smartcontract.Hash256Type:
+		return "interop.Hash256"
+	case smartcontract.PublicKeyType:
+		return "interop.PublicKey"
+	case smartcontract.SignatureType:
+		return "interop.Signature"
+	case smartcontract.ArrayType:
+		return "[]interface{}"
+	case smartcontract.MapType:
+		return "map[string]interface{}"
+	case smartcontract.InteropInterfaceType:
+		return "interface{}"
+	case smartcontract.VoidType:
+		return ""
+	default:
+		panic("unreachable")
+	}
+}
+
+// contractGenerateWrapper deploys contract.
+func contractGenerateWrapper(ctx *cli.Context) error {
+	m, _, err := readManifest(ctx.String("manifest"))
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("can't read contract manifest: %w", err), 1)
+	}
+
+	s, err := Generate(ctx, m)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during generation: %w", err), 1)
+	}
+
+	err = ioutil.WriteFile(ctx.String("out"), []byte(s), os.ModePerm)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("error during write: %w", err), 1)
+	}
+	return nil
+}
+
+func parseCallFlags(ctx *cli.Context) (map[string]string, error) {
+	flags := ctx.StringSlice("callflag")
+	result := make(map[string]string, len(flags))
+loop:
+	for i := range flags {
+		ss := strings.SplitN(flags[i], "=", 2)
+		if len(ss) == 1 {
+			return nil, fmt.Errorf("invalid callflag spec: %s", flags[i])
+		}
+		for i := range callFlags {
+			if ss[1] == callFlags[i] {
+				result[ss[0]] = ss[1]
+				continue loop
+			}
+		}
+		return nil, fmt.Errorf("invalid callflag spec: %s", flags[i])
+	}
+	return result, nil
+}
+
+type override struct {
+	Package  string
+	TypeName string
+}
+
+func parseOverrides(ctx *cli.Context) (map[string]override, error) {
+	overrides := ctx.StringSlice("override")
+	result := make(map[string]override, len(overrides))
+	for i := range overrides {
+		ss := strings.SplitN(overrides[i], "=", 2)
+		if len(ss) != 2 {
+			return nil, fmt.Errorf("invalid override spec: %s", overrides[i])
+		}
+
+		var over override
+		index := strings.LastIndexByte(ss[1], '.')
+		if index == -1 {
+			over.TypeName = ss[1]
+		} else {
+			over.Package = ss[1][:index]
+			switch over.Package {
+			case "iterator", "storage":
+				over.Package = "github.com/nspcc-dev/neo-go/pkg/interop/" + over.Package
+			case "ledger":
```

Comment:
Maybe then support also shorts for management's types? (manifest, contract).

Reply:
Fixed.